final-project/manApiProxy.tex

```diff
@@ -5,4 +5,11 @@ \chapter{REST API Proxy \\
 
 
 TODO: describe how REST API requests will be forwarded from the load balancers to the OTT monoliths
+
+When a REST API request is made to the OpenTogetherTube system, it is first received by the NGINX reverse proxy server. 
+NGINX then forwards the request to one of the OTT monoliths based on the current load balancing algorithm in use. 
+The monolith that receives the request will then process it and return a response, which will be sent back to the 
+NGINX reverse proxy server. The reverse proxy server will then forward the response back to the client that made the 
+original request.
```

Comment:
nit: nginx is usually all lowercase.

It's not necessarily nginx that is the reverse proxy. It can also be heroku or cloudflare, or in the current official site configuration cloudflare followed by heroku. These services are effectively stateless, so we can assume they are transparent, as in any request to any reverse proxy yields the same result.

Right now, if we take into consideration all of the components involved in serving a page from opentogethertube.com, you would observe the following. Note that this is not the only possible configuration (people self-host OTT).

```mermaid
flowchart LR
    Cloudflare --> Heroku --> OTT
```
With the smart load balancer it would look like this:
```mermaid
flowchart LR
    Cloudflare --> Heroku --> Balancer

    A[Monolith]
    B[Monolith]
    C[Monolith]

    Balancer --> A
    Balancer --> B
    Balancer --> C
```

So, you need to specify how the balancer is going to route requests to monoliths.